src/services/walletconnect/useWalletConnectClipboardUri.ts

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from 'react'
+import type { SetStateAction, Dispatch } from 'react'
+
+import { getClipboard, isClipboardGranted } from '@/utils/clipboard'
+
+export const useWalletConnectClipboardUri = (): [string, Dispatch<SetStateAction<string>>] => {
+  const [state, setState] = useState('')
+
+  useEffect(() => {
+    if (!navigator || !window) {
+      return
+    }
+
+    // Errors handled in the clipboard utils
+    const setClipboard = () => {
+      isClipboardGranted().then((isGranted) => {
+        if (isGranted) {
+          getClipboard().then(setState)
+        }
+      })
+    }
```

Comment:
This needs to check if it starts with `wc:`, otherwise it will open the popup for any copied text.

Also please make it an async function.

Reply:
There [is a check in `WcInput`](https://github.com/safe-global/safe-wallet-web/blob/focus-wc-uri/src/components/walletconnect/WcInput/index.tsx#L46) but I added one to the hook as well.